initial fastlane structures from IzzyOnDroid

IzzySoft opened the pull request:
this PR provides you with a "fastlane starter package" – to give it into your hands to define how your app is presented, and make it easy to keep descriptions & graphics in sync with your development. Some notes to sum them up in a place easy to find for you:

* the [IzzyOnDroid Fastlane Documentation](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Fastlane) can guide you with maintaining the structures here
* take special care for limits, e.g. per-release changelogs can have max 500 chars, `full_description.txt` max 4,000 chars, graphics must have specific aspect ratios etc.
* you can have "bigger graphics" here; the IoD updater takes care to resize/optimize them as needed
* for the `full_description.txt` here I've used HTML compressed into a single line, to prevent the fdroid software converting each line break to a `<br>`. The tags used for this are supported by F-Droid.org as well, and to my knowledge even by PlayStore.
* whenever you add a new "type" of metadata that has not been there before (e.g. the first changelog, a featureGraphic if it wasn't yet present), please let us know so we can check and enable it.
* Details on all this in the linked documentation.
* Once merged, your changes will be synced whenever a new release is pulled in here – to make sure it stays up-to-date with the app itself.
* Should you have questions, be welcome to ask :wink:

The second commit adds the link to IzzyOnDroid to the Readme. Feel free to cherry-pick or squash :wink:

And now: Enjoy!

IzzySoft commented:
Thanks for merging – and enhancing even :hugs: Small note: the changelogs must be named `<versionCode>.txt` to be recognized, you've used `versionName` there.

All else is activated now on our end – i.e. whatever you change from now on, should be picked up whenever a new release is being pulled in. Seeing you've updated the files already, I manually triggered an update now, so those should show up with the next sync (around 6 pm UTC) already. Please let me know once changelogs are corrected, so I can check and enable them as well.

scheibler commented:
Aha, you mean the Android version code - in this case "12"?
Therefore: `.../changelogs/12.txt`?

That was the only thing I wondered about. You had 100100.txt in your example and I didn't know if that should be the
version name without . or - or if it should be something else.

Thanks for clarification.

IzzySoft commented:
> Aha, you mean the Android version code - in this case "12"?
> Therefore: `.../changelogs/12.txt`?

Exactly!

> That was the only thing I wondered about. You had 100100.txt in your example and I didn't know if that should be the version name without . or - or if it should be something else.

Let me check my example. If I can improve that to be more helpful, I gladly do that!

```
            │   └── changelogs              ³
            │       ├── 100000.txt          (must correspond to versionCode, literally, no padding)
            │       ├── 100100.txt          (so this means: versionCode=100100)
            │       └── 100101.txt          (max size: 500 bytes, plain ASCII (no HTML))
```

Hm, it explicitly says "must correspond to versionCode, literally, no padding" and "so this means: versionCode=100100". Any suggestion how to make that more clear? Using only 2-digit examples? That one was taken from an app that based its `versionCode` on its `versionName`, to keep them in sync – so indeed, `100101.txt` was `versionCode: 100101, versionName: "1.1.1"` – calculating its `versionCode` as `major x 10000 + minor x 100 + patch`, going by [semantic versioning](https://semver.org/). I was hoping the explanations next to the examples were clear enough …

scheibler commented:
> 100100.txt          (so this means: versionCode=100100)

Yes, if you read carefully you can figure it out...

But for those like me, maybe something like:

> 100100.txt          (so this means: versionCode=100100, e.g. from app/build.gradle)

could make it even more obvious.

I also was not familiar with the "versioncode represents versionname" schema. The projects I created or worked with
had much lower int numbers for the versioncode attribute and were not linked that way.

So I wondered about the version update from 100000 to 100100 and thought "maybe the version name without '.'s?"...

Therefore: If I had read your comment more thoroughly or were familiar with the versioncode scheme it would have been fine :-).

IzzySoft commented:
> Yes, if you read carefully you can figure it out...

Yupp. But small things are easily overlooked. Happens to me, too, and more often than I'd wish…

> But for those like me, maybe something like:

Thanks for your suggestion! Implemented right away :hugs: And live already.

> I also was not familiar with the "versioncode represents versionname" schema.

There are so many variants on how `versionCode` is dealt with. It's an integer that must be increased with each new release – but no norms by how much. Some simply use it as "release counter" (starting with 1 and incrementing by 1 with each release). Others as release date indicator (`YYYYMMDD[HH]` or the likes). Again others try to match it to `versionName` somehow…

> Therefore: If I had read your comment more thoroughly or were familiar with the versioncode scheme it would have been fine :-).

Hehe, no worries. That's why we check before enabling, so such mistakes do not go unnoticed (with you wondering at some point why your changelogs never show up).

So: added that at our end as well now that this is fine, refreshed our local clone, and synced fastlane with our metadata. Changelogs are now picked up as well, and `12.txt` takes effect with the next repo sync around 6 pm UTC. Thanks a lot!

scheibler commented:
Sorry to bother you again but it seems that I got the same problem with upcoming releases like in my other project: https://github.com/scheibler/WalkersGuide-Android/pull/9

At least it also doesn't pick up my new releases. I published version 2.5.0 12 days ago and v2.5.1 today but your repo still lists 2.4.0 as the latest version.

Do you know why that could be? Thanks again.

IzzySoft commented:
> Do you know why that could be?

Yupp: there was a huge gap in releases between v2.3.0 and v2.4.0 (more than a year), so the app was demoted to the monthly update track. After 2.4.0, it was again over a year until v2.5.0, so it was kept in the monthly track. After that, there were just 2 recent releases now (with the last just hours ago), so it was not yet pushed back to the daily track. I did that now manually (putting it back to the daily track), and also manually triggered an update, so v2.5.1 was just pulled in.

scheibler commented:
Cool! After so many changes to this small project I think, the next release will take more than a year again, so that
your algorithm can do its magic and I bother you again in 2027 ;-).

Thanks for checking - at least it was apparently not the other issue again!

IzzySoft commented:
Glad it was an easy one this time, yeah. Should there be no new release within a year, it might be the same one next time then :zany_face:
